platform/features/layout/src/FixedController.js

```diff
@@ -311,26 +288,228 @@ define(
+         */
+        FixedController.prototype.updateView = function (telemetryObject, datum) {
+            var metadata = this.openmct.telemetry.getMetadata(telemetryObject);
+            var rangeMetadata = metadata.valuesForHints(['range'])[0];
+            var rangeKey = rangeMetadata.source || rangeMetadata.key;
+            var valueMetadata = metadata.value(rangeKey);
+            var limitEvaluator = this.openmct.telemetry.limitEvaluator(telemetryObject);
+            var formatter = this.openmct.telemetry.getValueFormatter(valueMetadata);
+            var value = datum[valueMetadata.key];
+            var alarm = limitEvaluator && limitEvaluator.evaluate(datum, rangeKey);
+
+            this.setDisplayedValue(
+                telemetryObject,
+                formatter.format(value),
+                alarm && alarm.cssClass
+            );
+            this.digest();
+        };
+
+        /**
+         * Returns a new or cached telemetry collection associated with a given
+         * object.
+         * @private
+         * @param {object} object A domain object
+         * @returns {TelemetryCollection} the collection of telemetry for the given object
+         */
+        FixedController.prototype.getCollectionForObject = function (object) {
+            var telemetryCollection = this.telemetryCollections.get(object);
+            var timeSystem;
+            var domainKey;
+            var metadata;
+            var domainMetadata;
+            var self = this;
+
+            if (telemetryCollection === undefined) {
+                timeSystem = this.openmct.conductor.timeSystem();
+                metadata = this.openmct.telemetry.getMetadata(object);
+                domainMetadata = metadata.valuesForHints(['domain']).filter(function (value) {
+                    return value.key === timeSystem.metadata.key;
+                })[0];
+
+                if (domainMetadata) {
+                    domainKey = domainMetadata.source || domainMetadata.key;
+                } else {
+                    domainKey = timeSystem.metadata.key;
+                }
+
+                telemetryCollection = new TelemetryCollection();
+                telemetryCollection.bounds(this.openmct.conductor.bounds());
+                telemetryCollection.sort(domainKey);
+                telemetryCollection.on("added", function (telemetryAdded) {
+                    var datum = telemetryAdded[telemetryAdded.length - 1];
+                    if (datum !== undefined) {
+                        self.updateView(object, datum);
+                    }
+
+                    // Optimization, clear the cached telemetry to free
+                    // up memory. We're only interested in the last value <
+                    // bounds.end
+                    telemetryCollection.telemetry = [];
+                });
+
+                this.telemetryCollections.set(object, telemetryCollection);
```

Comment:
Think this may allow redundant subscriptions if different instances of the same object end up getting passed into `getCollectionForObject`. Could this be done with a plain object instead of a Map and key-strings instead?